# Import Libraries

In [1]:
import numpy as np
import pandas as pd

# Loading Dataset

In [2]:
df = pd.read_excel("E:/project1_of_decodelab/Dataset for Data Analytics.xlsx")

# Initial Inspection

In [3]:
df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [4]:
df.shape

(1200, 14)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(

In [6]:
df.describe()

,Date,Quantity,UnitPrice,ItemsInCart,TotalPrice
count,1200,1200.000000,1200.000000,1200.000000,1200.000000
mean,2024-03-22 16:58:48,2.945833,356.412750,5.485000,1053.968300
min,2023-01-01 00:00:00,1.000000,11.390000,1.000000,11.390000
25%,2023-08-03 18:00:00,2.000000,186.062500,4.000000,410.520000
50%,2024-03-23 00:00:00,3.000000,364.210000,5.000000,823.615000
75%,2024-11-08 12:00:00,4.000000,521.570000,7.000000,1578.475000
max,2025-06-30 00:00:00,5.000000,699.930000,10.000000,3456.400000
std,NaN,1.407557,197.177146,2.281983,819.856558


### There is 309 missing values in CouponCode column so i just fill it with No coupon

In [10]:
df['CouponCode'] = df['CouponCode'].fillna("No Coupon")

### Checkin the unique total orders

In [11]:
total_orders = df['OrderID'].nunique()
print("Total Orders",total_orders)

Total Orders 1200


### Checking Total Revenue 

In [12]:
total_revenue = df['TotalPrice'].sum()
print("Total revenue:",total_revenue)

Total revenue: 1264761.96


### Average of value of Orders

In [14]:
average_order_value = df['TotalPrice'].mean()
print("Average value of orders",average_order_value)

Average value of orders 1053.9683


### Checking Maximum and Minimum Order value

In [15]:
max_order_value = df['TotalPrice'].max()
min_order_value = df['TotalPrice'].min()
print("Maximum value of the order",max_order_value)
print("Minimum value of the order",min_order_value)

Maximum value of the order 3456.4
Minimum value of the order 11.39


### Checking which product generate a large amount of revenue

In [15]:
df.groupby('Product')['TotalPrice'].sum()

Product
Chair      195620.11
Desk       167459.93
Laptop     192126.56
Monitor    175651.41
Phone      151722.39
Printer    195612.61
Tablet     186568.95
Name: TotalPrice, dtype: float64

### Highest and Lowest Revenue Product

In [16]:

product_revenue = df.groupby('Product')['TotalPrice'].sum()


top_product = product_revenue.idxmax()
top_revenue = product_revenue.max()

lowest_product = product_revenue.idxmin()
lowest_revenue = product_revenue.min()

print("--- Product Revenue Analysis ---")
print(f"Top Revenue Product: {top_product} (Rs. {top_revenue:,.2f})")
print(f"Lowest Revenue Product: {lowest_product} (Rs. {lowest_revenue:,.2f})")

--- Product Revenue Analysis ---
Top Revenue Product: Chair (Rs. 195,620.11)
Lowest Revenue Product: Phone (Rs. 151,722.39)


### Checking Payment Methods

In [17]:
df['PaymentMethod'].value_counts()

PaymentMethod
Online         258
Cash           246
Credit Card    234
Debit Card     232
Gift Card      230
Name: count, dtype: int64

### Checking Order Status

In [18]:
df['OrderStatus'].value_counts()

OrderStatus
Cancelled    250
Returned     247
Pending      237
Shipped      235
Delivered    231
Name: count, dtype: int64

### Checking Unique CouponCode

In [18]:
df['CouponCode'].value_counts()

CouponCode
FREESHIP     313
No Coupon    309
WINTER15     292
SAVE10       286
Name: count, dtype: int64

### Checking impact of CouponCode

In [19]:

df['Coupon_Status'] = df['CouponCode'].apply(lambda x: 'Without Coupon' if x == 'No Coupon' else 'With Coupon')


coupon_comparison = df.groupby('Coupon_Status').agg(
    Total_Orders=('OrderID', 'nunique'),
    Total_Revenue=('TotalPrice', 'sum'),
    Average_Order_Value=('TotalPrice', 'mean'),
    Average_Quantity=('Quantity', 'mean')
).round(2) 


print("--- Coupon Analysis: With Coupon vs Without Coupon ---")
print(coupon_comparison)

--- Coupon Analysis: With Coupon vs Without Coupon ---
                Total_Orders  Total_Revenue  Average_Order_Value  \
Coupon_Status                                                      
With Coupon              891      942360.55              1057.64   
Without Coupon           309      322401.41              1043.37   

                Average_Quantity  
Coupon_Status                     
With Coupon                 2.91  
Without Coupon              3.05  


### Checking the Source

In [20]:
df['ReferralSource'].value_counts()

ReferralSource
Instagram    259
Email        250
Google       241
Facebook     228
Referral     222
Name: count, dtype: int64

### Detecting Outliers

In [21]:

def check_outliers(column):
    
    Q1 = df[column].quantile(0.25)
    Q3 = df[column].quantile(0.75)
    IQR = Q3 - Q1
    
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR
    
    outliers = df[(df[column] < lower_bound) | (df[column] > upper_bound)]
    

    print(f"--- Outliers Analysis for {column} ---")
    print(f"Normal Range: {lower_bound:.2f} to {upper_bound:.2f}")
    print(f"Total Outliers Detected: {len(outliers)}\n")
    
    return outliers


price_outliers = check_outliers('TotalPrice')
qty_outliers = check_outliers('Quantity')



--- Outliers Analysis for TotalPrice ---
Normal Range: -1341.41 to 3330.41
Total Outliers Detected: 8

--- Outliers Analysis for Quantity ---
Normal Range: -1.00 to 7.00
Total Outliers Detected: 0



### Revenue Analysis

* Total Revenue: 1,264,761.96
* Average Order Value: 1,053.29
* Highest Order Value: 3,456.40
* Lowest Order Value: 11.39

### Product Performance

* Chair generated the highest revenue.
* Desk ranked second.
* Laptop ranked third.
* Phone generated the lowest revenue.

### Payment Method Analysis

* Online Payment was the most frequently used method.
* Cash, Credit Card, and Debit Card showed similar usage patterns.
* Customers demonstrated a preference for digital payments.

### Order Status Analysis

* Cancelled Orders: 250
* Returned Orders: 247

High cancellation and return rates suggest potential issues related to customer satisfaction, order fulfillment, or delivery processes.

### Coupon Analysis

* Orders with Coupons: 891
* Orders without Coupons: 309

Revenue Generated:

* Coupon Users: 942,360
* Non-Coupon Users: 322,401

Coupon campaigns significantly increased customer spending and overall revenue.

### Referral Source Analysis

Top customer acquisition channels:

1. Social Media
2. Email Marketing
3. Google
4. Facebook
5. Referral Traffic

The business benefits from diversified marketing channels.